### PR TITLE
Non-numpy-wrapped image payloads

### PR DESCRIPTION
**Pull request by athewsey**
**Issue #, if available:** #7

**Description of changes:**

- Use DataSerializer instead of the default NumpySerializer when invoking the endpoint, to avoid the JPEG bytes getting wrapped in a npy format.
- Move the pip install ahead of any Python `import`s as a general best-practice, because importing libraries before installs can cause caching issues in case any of the imported library versions are affected by the installs.
- Prefer `%pip install` over `!pip3 install`, as the former should be guaranteed to always use the pip/Python environment that the current notebook kernel is running - whereas the latter may sometimes differ.

<br/>
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: sm-notebook/code/inference.py [hunk 1]
```diff
@@ -38,4 +47,4 @@ def output_fn(prediction_output, content_type):
             infer['keypoints'] = result.keypoints.numpy().data.tolist()
         if 'probs' in result.keys:
             infer['probs'] = result.probs.numpy().data.tolist()
-    return json.dumps(infer)
\ No newline at end of file
+    return json.dumps(infer)
```
Comment (rromanchuk) [inline sm-notebook/code/inference.py:50]:
```suggestion
def output_fn(prediction, content_type):
    print("Executing output_fn from inference.py ...")
    return prediction[0].tojson()
```

If i was consuming this endpoint i would appreciate response that didn't have me questioning the sanity of ~~python~~ tensor programmers 🤣 

[payload](https://docs.ultralytics.com/hub/inference_api/#return-json-format)
```
{
  "success": True,
  "message": "Inference complete.",
  "data": [
    {
      "name": "person",
      "class": 0,
      "confidence": 0.8359682559967041,
      "box": {
        "x1": 0.08974208831787109,
        "y1": 0.27418340047200523,
        "x2": 0.8706787109375,
        "y2": 0.9887352837456598
      }
    },
    {
      "name": "person",
      "class": 0,
      "confidence": 0.8189555406570435,
      "box": {
        "x1": 0.5847355842590332,
        "y1": 0.05813225640190972,
        "x2": 0.8930277824401855,
        "y2": 0.9903111775716146
      }
    },
    {
      "name": "tie",
      "class": 27,
      "confidence": 0.2909725308418274,
      "box": {
        "x1": 0.3433395862579346,
        "y1": 0.6070465511745877,
        "x2": 0.40964522361755373,
        "y2": 0.9849439832899306
      }
    }
  ]
}
```

File: sm-notebook/code/inference.py [hunk 2]
```diff
@@ -2,30 +2,39 @@
...
     return img
-    
+
+
 def predict_fn(input_data, model):
     print("Executing predict_fn from inference.py ...")
     device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
     model.to(device)
     with torch.no_grad():
         result = model(input_data)
```
Comment (rromanchuk) [inline sm-notebook/code/inference.py:34]:
```suggestion
        result = model(input_data, imgsz=[300, 300])
```

Default is `imgsz=640`,  but implied in test notebook, hints the model is explicitly trained for 300x300,  or some other reason.  I was confused why there was downscaling before inference, so make it explicit to align with example.  This still would get preprocessed because its not a stride of 32 so pointless on client/server and double pointless on client. :)

File: sm-notebook/2_TestEndpoint.ipynb
Comment (rromanchuk) [inline sm-notebook/2_TestEndpoint.ipynb:1]: Maybe there is good reason, but this seems way over complicated to me.  YOLO already does this this preprocessing and returns source information in `orig_shape`, if you need it for some reason. 

```python

infer_start_time = time.time()

orig_image = cv2.imread('bus.jpg')

image_height, image_width, _ = orig_image.shape
model_height, model_width = 300, 300
x_ratio = image_width/model_width
y_ratio = image_height/model_height

resized_image = cv2.resize(orig_image, (model_height, model_width))
payload = cv2.imencode('.jpg', resized_image)[1].tobytes()
result = predictor.predict(payload)

infer_end_time = time.time()

print(f"Inference Time = {infer_end_time - infer_start_time:0.4f} seconds")
```

Just send the original source that the model, was supposedly already trained to expect,  and you get back the normalized bbox on supplied image. 

```python

infer_start_time = time.time()

orig_image = cv2.imread('bus.jpg')
payload = cv2.imencode('.jpg', orig_image)[1].tobytes()
result = predictor.predict(payload)

infer_end_time = time.time()

print(f"Inference Time = {infer_end_time - infer_start_time:0.4f} seconds")
```

draw box, label, confidence on your source, as is, regardless of source dimensions 

```python
for result in results:
    box = result["box"]
    lbl = result["name"]
    conf = result["confidence"]
    x1, y1, x2, y2 = int(box["x1"]), int(box["y1"]), int(box["x2"]), int(box["y2"])
    color = (random.randint(10,255), random.randint(10,255), random.randint(10,255))
    cv2.rectangle(orig_image, (x1,y1), (x2,y2), color, 4)
    cv2.putText(orig_image, f"Class: {lbl}", (x1,y1-40), cv2.FONT_HERSHEY_SIMPLEX, 1, color, 2, cv2.LINE_AA)
    cv2.putText(orig_image, f"Conf: {int(conf*100)}", (x1,y1-10), cv2.FONT_HERSHEY_SIMPLEX, 1, color, 2, cv2.LINE_AA)
```

**Review by rromanchuk — CHANGES_REQUESTED** [review]
Really just notes for aws, piggy backing off your PR.

ps: the CF yaml linked on the blog is not launch-able because of unsupported node lambda runtime
